src/launch/types.rs

```diff
@@ -54,6 +54,32 @@ impl<'a> LaunchUpdateData<'a> {
     }
 }
 
+/// Inject a secret into the guest.
+#[repr(C)]
+pub struct LaunchSecret<'a> {
+    hdr_addr: u64,
+    hdr_len: u32,
+    guest_addr: u64,
+    guest_len: u32,
+    trans_addr: u64,
+    trans_len: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> LaunchSecret<'a> {
+    pub fn new(header: &'a Header, guest: &u8, trans: &'a [u8]) -> Self {
```

Comment:
```suggestion
    pub fn new(header: &'a Header, guest: usize, trans: &'a [u8]) -> Self {
```

`&u8` is probably not a good way to express a Guest Physical Address. I wish we had a good type for this.